packages/vscode/src/plugins/nextjs/addJSDoc.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  ArrowFunction,
+  FunctionDeclaration,
+  SourceFile,
+  TypeGuards,
+} from 'ts-morph'
+import { NextFunctionName, NextFunctionType } from './constants'
+
+function buildJSDocType(base: string, paramName: string) {
+  return `{import('next').${base} } ${paramName}`
+}
+function addType(
+  node: ArrowFunction | FunctionDeclaration,
+  foundNextFunctions: NextFunctionType,
+) {
+  const functionNames = Object.keys(foundNextFunctions) as NextFunctionName[]
+  if (functionNames && functionNames.length !== 0) {
+    const params = node.getParameters()
```

Comment:
There needs to be a check for the length of params before accessing the first element.

Reply:
```
const blah = []
if(blah[0]){}
```